### PR TITLE
[DOC] Enhanced RDoc for IO

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Revises much of the introductory material.

Also treats:
- #close_on_exec
- #close
- #closed?
- #close_read
- #close_write

### CODE REVIEW

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks mostly good, only a few minor changes requested.

File: io.c [hunk 1]
```diff
@@ -4951,7 +4951,7 @@ rb_io_close_on_exec_p(VALUE io)
 #if defined(HAVE_FCNTL) && defined(F_GETFD) && defined(F_SETFD) && defined(FD_CLOEXEC)
 /*
  *  call-seq:
- *     ios.close_on_exec = bool    -> true or false
+ *    close_on_exec = bool -> true or false
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply io.c]: Done.

File: io.c [hunk 2]
```diff
@@ -13751,82 +13735,56 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
...
- *    operating system.
+ *  An instance of class \IO (commonly called a _stream_)
+ *  represents an input/output stream in the underlying operating system.
+ *  \Class \IO is the basis for input and output in Ruby.
  *
- *  * A string starting with <code>"|"</code> indicates a subprocess.
- *    The remainder of the string following the <code>"|"</code> is
- *    invoked as a process with appropriate input/output channels
- *    connected to it.
+ *  \Class File is the only class in the Ruby core that derives from \IO.
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply io.c]: Done.

File: io.c [hunk 3]
```diff
@@ -13751,82 +13735,56 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
...
- *
- *  Example:
- *
- *    require 'io/console'
- *    rows, columns = $stdout.winsize
- *    puts "Your screen is #{columns} wide and #{rows} tall"
+ *  Extension <tt>io/console</tt> provides numerous methods
+ *  for interacting with the console;
+ *  requiring it adds numerous methods to class \IO.
+ *  See {io/console}[https://ruby-doc.org/stdlib/libdoc/io/console/rdoc].
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply io.c]: Removed.

**Review by jeremyevans — APPROVED** [review]
Looks good!

**Pull request merged**